applications/crypto/config.ml

```diff
@@ -12,5 +12,5 @@ let packages =
     package ~min:"0.2.0" "randomconv";
   ]
 
-let main = main "Unikernel.Main" ~packages (random @-> job)
-let () = register "crypto-test" [ main $ default_random ]
+let main = main "Unikernel" ~packages ~deps:[ dep noop ] job
```

Comment:
Why is the `dep noop` needed here?

Reply:
to delay the execution of start (now having a unit argument).

what happens if you forget it is that mirage configure will complain (see https://github.com/mirage/mirage/pull/1544) -- certainly we should adapt the error message :)

Reply:
is there any useful unikernel/app that could be written with `main "Unikernel" ~packages job`? If not, should we just insert it that `noop` by default in that case (with a warning message to make clear that some magic has happened)?  I'm just a bit concerned that this makes the "hello world" unikernel unnecessary harder to write (as you have to explain what `deps` is, what `noop` is, etc).

Or maybe we should have a `noop -> job` signature that just adds a `()` argument to the main function without changing the functor structure.  That seems a bit more convoluted though.

Reply:
sure, we can always add a () argument to the job of main. I'd make this unconditionally, and not "only if there's no other dep/functor). Now that's an orthogonal change requiring mirage/functoria insights and needs to be tested with mirage-www/dns-primary-git unikernels (esp. with --enable-monitoring variant).

Please feel free to provide a PR that does this in the mirage/mirage repository. I'm very low on functoria skills.

Reply:
With https://github.com/mirage/mirage/pull/1599/commits/c92714f934592ff8ff61bc81028fe77235ed158a this does emit it directly. I'll push a commit here to show the changes.

Reply:
and here 0d0c5e528e21aefe9ddfcfc9f3c7e3d10f47e8c1

Reply:
For reference here is some previous discussion about this topic https://github.com/mirage/mirage/issues/1426